### PR TITLE
refactor: make `rich-text` not based on model

### LINKED ISSUE #4325: Markdown shortcut sometimes not works
https://github.com/toeverything/blocksuite/assets/50035259/3778cc43-2c03-45d5-9b0a-14270c5c1262

### PR DESCRIPTION
**Pull request by Flrande**
Close #4325

### CODE REVIEW

**Comment by zzj3720** [comment]
lgtm

File: packages/blocks/src/__internal__/rich-text/keymap/container.ts [hunk 1]
```diff
@@ -66,10 +192,87 @@ export const bindContainerHotkey = (blockElement: BlockElement) => {
...
+        assertExists(vRange);
+        hardEnter(model, vRange, vEditor, state.raw, true);
+        _preventDefault(ctx);
+
+        return true;
+      }
+
+      return;
+    },
+    ' ': ctx => {
```
Comment (Saul-Mirone) [inline packages/blocks/src/__internal__/rich-text/keymap/container.ts:261]: You can also use `Space`

**Review by Saul-Mirone — APPROVED** [review]
Great work!

**Review by Saul-Mirone — CHANGES_REQUESTED** [review]
Seems the behavior of `ArrowUp` and `ArrowDown` is not right in multi line rich text

**Review by Saul-Mirone — APPROVED** [review]

**Comment by Flrande** [comment]
It will be merged after AFFiNE stable version released.

**Pull request merged**